spec/script/create_written_spec.rb

```diff
@@ -0,0 +1,121 @@
+require Rails.root.join('script', 'create_written.rb').to_s
+
+RSpec.describe "#create_writtens" do # rubocop:disable Rspec/DescribeClass
+  it "skips posts that already have writtens" do
+    create(:post)
+    expect { create_writtens }.not_to change{Reply.count}
+  end
+
+  it "creates writtens for simple posts" do
+    post = create(:post, skip_written: true)
+    expect(post.written).to be_nil
+    allow(STDOUT).to receive(:puts).with("Creating writtens for posts #{post.id} through #{post.id}")
+    Timecop.freeze(Time.zone.now + 8.hours) do
+      expect { create_writtens }.to change{Reply.count}.by(1)
+    end
+    written = Reply.last
+    expect(written.reply_order).to eq(0)
+    expect(written.user).to eq(post.user)
+    expect(written.content).to eq(post.content)
+    expect(written.icon_id).to be_nil
+    expect(written.character_id).to be_nil
+    expect(written.character_alias_id).to be_nil
+    expect(written.created_at).to be_the_same_time_as(post.created_at)
+    expect(written.updated_at).to be_the_same_time_as(post.edited_at)
+  end
+
+  it "creates correct writtens for complex posts" do
+    user = create(:user)
+    icon = create(:icon, user: user)
+    character = create(:template_character, default_icon: icon, user: user)
+    calias = create(:alias, character: character)
+
+    post = create(:post, character: character, icon: icon, character_alias: calias, user: user, skip_written: true)
+    Timecop.freeze(Time.zone.now + 4.hours) do
+      post.update!(content: 'new content!')
+    end
+    expect(post.written).to be_nil
+
+    allow(STDOUT).to receive(:puts).with("Creating writtens for posts #{post.id} through #{post.id}")
+    Timecop.freeze(Time.zone.now + 8.hours) do
+      expect { create_writtens }.to change{Reply.count}.by(1)
+    end
+    written = Reply.last
+    expect(written.reply_order).to eq(0)
+    expect(written.user).to eq(post.user)
+    expect(written.content).to eq(post.content)
+    expect(written.icon_id).to eq(icon.id)
+    expect(written.character_id).to eq(character.id)
+    expect(written.character_alias_id).to eq(calias.id)
+    expect(written.created_at).to be_the_same_time_as(post.created_at)
+    expect(written.updated_at).to be_the_same_time_as(post.edited_at)
+  end
+
+  it "handles audits correctly" do
+    Post.auditing_enabled = true
```

Comment:
Can we put these audit things into a `before(:each)` and `after(:each)`, so a failure here keeps the state changes local? (It might actually need a `around(:each)` to ensure it puts it back after a failure, not sure, but I think we've done a similar thing elsewhere with audits)

Reply:
Huh. I can do that, yeah.